arch/risc-v/src/common/riscv_fork.c

```diff
@@ -147,15 +149,15 @@ pid_t riscv_fork(const struct fork_s *context)
   /* Set up frame for context and copy the parent's user context there */
 
   memcpy((void *)(newsp - XCPTCONTEXT_SIZE),
```

Comment:
The error is here, newsp is using the child's _user stack_ to store the context. It should use the kernel stack.

Also, the fork() error is not related to this PR whatsoever, please make it into a separate patch and a separate PR.

Reply:
SMP mode fork is also broken due to #12812. When the child runs, the context is read from the user stack but the wrong mappings are in use -> the child crashes due to page fault.

Please, let me fix the fork() issue @hujun260 @lupyuen